File: Components/ASP.NET/VB/ASPxImageAndDataNavigationDemos/App_Code/BinaryImagesModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data.Entity
Imports System.Data.Entity.Infrastructure
Imports System.Data.Entity.ModelConfiguration

Partial Public Class BinaryImagesContext
	Inherits ContextBase
	Shared Sub New()
		Database.SetInitializer(Of BinaryImagesContext)(Nothing)
	End Sub

	Public Sub New()
		MyBase.New("BinaryImagesConnectionString")
	End Sub

	Public Property BinaryImages() As DbSet(Of BinaryImage)

	Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
		modelBuilder.Configurations.Add(New BinaryImageMap())
	End Sub
End Class

Partial Public Class BinaryImage
	Public Property ID() As Integer
	Public Property Text() As String
	Public Property Picture() As Byte()
	Public Property CategoryID() As Integer?
End Class

Public Class BinaryImageMap
	Inherits EntityTypeConfiguration(Of BinaryImage)
	Public Sub New()
		' Primary Key
		Me.HasKey(Function(t) t.ID)

		' Properties
		Me.Property(Function(t) t.ID)

		Me.Property(Function(t) t.Text).IsRequired().HasMaxLength(50)

		' Table & Column Mappings
		Me.ToTable("BinaryImages")
		Me.Property(Function(t) t.ID).HasColumnName("ID")
		Me.Property(Function(t) t.Text).HasColumnName("Text")
		Me.Property(Function(t) t.Picture).HasColumnName("Picture")
		Me.Property(Function(t) t.CategoryID).HasColumnName("CategoryID")
	End Sub
End Class

File: Components/ASP.NET/CS/MVCxImageAndDataNavigationDemos/Models/BinaryImagesModel.cs
using System;
using System.Data.Entity;
using System.Data.Entity.Infrastructure;
using System.Data.Entity.ModelConfiguration;

public partial class BinaryImagesContext : ContextBase {
    static BinaryImagesContext() {
        Database.SetInitializer<BinaryImagesContext>(null);
    }

    public BinaryImagesContext() : base("BinaryImagesConnectionString") { }

    public DbSet<BinaryImage> BinaryImages { get; set; }

    protected override void OnModelCreating(DbModelBuilder modelBuilder) {
        modelBuilder.Configurations.Add(new BinaryImageMap());
    }
}

public partial class BinaryImage {
    public int ID { get; set; }
    public string Text { get; set; }
    public byte[] Picture { get; set; }
    public Nullable<int> CategoryID { get; set; }
}

public class BinaryImageMap : EntityTypeConfiguration<BinaryImage> {
    public BinaryImageMap() {
        // Primary Key
        this.HasKey(t => t.ID);

        // Properties
        this.Property(t => t.ID);

        this.Property(t => t.Text)
            .IsRequired()
            .HasMaxLength(50);

        // Table & Column Mappings
        this.ToTable("BinaryImages");
        this.Property(t => t.ID).HasColumnName("ID");
        this.Property(t => t.Text).HasColumnName("Text");
        this.Property(t => t.Picture).HasColumnName("Picture");
        this.Property(t => t.CategoryID).HasColumnName("CategoryID");
    }
}
